src/blocks/privacy.rs

```diff
@@ -0,0 +1,201 @@
+//! Privacy Monitor
+//!
+//! # Configuration
+//!
+//! Key        | Values | Default|
+//! -----------|--------|--------|
+//! `driver` | The configuration of a driver (see below). | **Required**
+//! `format`   | Format string. | <code>"{ $icon_audio \|}{ $icon_audio_sink \|}{ $icon_video \|}{ $icon_webcam \|}{ $icon_unknown \|}"</code> |
+//! `format_alt`   | Format string. | <code>"{ $icon_audio $info_audio \|}{ $icon_audio_sink $info_audio_sink \|}{ $icon_video $info_video \|}{ $icon_webcam $info_webcam \|}{ $icon_unknown $info_unknown \|}"</code> |
+//!
+//! # pipewire Options (requires the pipewire feature to be enabled)
+//!
+//! Key | Values | Required | Default
+//! ----|--------|----------|--------
+//! `name` | `vl4` | Yes | None
+//! `exclude_device` | A device to ignore, example: `["/dev/video5"]` | No | `[]`
+//! `exclude_consumer` | Processes to ignore | No | `["pipewire", "wireplumber"]`
+//!
+//! # vl4 Options
+//!
+//! Key | Values | Required | Default
+//! ----|--------|----------|--------
+//! `name` | `vl4` | Yes | None
+//! `exclude_device` | A device to ignore, example: `["/dev/video5"]` | No | `[]`
+//! `exclude_consumer` | Processes to ignore | No | `["pipewire", "wireplumber"]`
+//!
+//! # Available Format Keys
+//!
+//! Placeholder                                      | Value                                          | Type     | Unit
+//! -------------------------------------------------|------------------------------------------------|----------|-----
+//! `icon_{audio,audio_sink,video,webcam,unknown}`   | A static icon                                  | Icon     | -
+//! `info_{audio,audio_sink,video,webcam,unknown}`   | The mapping of which source are being consumed | Text     | -
+//!
+//! You can use the suffixes noted above to get the following:
+//!
+//! Suffix       | Description
+//! -------------|------------
+//! `audio`      | Captured audio (ex. Mic)
+//! `audio_sink` | Audio captured from a sink (ex. openrgb)
+//! `video`      | Video capture (ex. screen capture)
+//! `webcam`     | Webcam capture
+//! `unknown`    | Anything else
+//!
+//! # Available Actions
+//!
+//! Action          | Description                               | Default button
+//! ----------------|-------------------------------------------|---------------
+//! `toggle_format` | Toggles between `format` and `format_alt` | Left
+//!
+//! # Example
+//!
+//! ```toml
+//! [[block]]
+//! block = "privacy"
+//! merge_with_next = true
```

Comment:
Do you think that we should allow one or more driver to be used by the same block instead of just using two and merging them?

Reply:
Maybe we don't need a `driver` option and can guess the driver based on format placeholders used? For example `format = " $icon_video |"` would use `v4l` and `pipewire` (if enabled and available). Driver-specific options can be added in the main config if needed.

Reply:
Also, wouldn't merging two privacy blocks result in potentially duplicated icons?

Reply:
`merge_with_next = true` could result in multiple camera icons if you have two different applications that access a camera through a different mechanism (v4l + pipewire).
If we allow multiple drivers to be used in the same block we can merge the results from all of the drivers.